src/types/traces.rs

```diff
@@ -0,0 +1,300 @@
+                let mut result = None;
+                let mut tx_type: Option<TxType> = None;
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::TraceAddress => {
+                            if trace_address.is_some() {
+                                return Err(de::Error::duplicate_field("trace_address"));
+                            }
+                            trace_address = Some(map.next_value()?); // Vec<usize>
+                        },
+                        Field::Subtraces => {
+                            if subtraces.is_some() {
+                                return Err(de::Error::duplicate_field("subtraces"));
+                            }
+                            subtraces = Some(map.next_value()?); // usize
+                        },
+                        Field::Action => {
+                            if action.is_some() {
+                                return Err(de::Error::duplicate_field("action"));
+                            }
+                            action = Some(map.next_value()?); // serde_json `Value`
+                        },
+                        Field::Result => {
+                            if result.is_some() {
+                                return Err(de::Error::duplicate_field("trace_result"));
+                            }
+                            result = Some(map.next_value()?); // Res
+                        },
+                        Field::TransactionType => {
+                            if tx_type.is_some() {
+                                return  Err(de::Error::duplicate_field("transaction_type"));
+                            }
+                            tx_type = Some(map.next_value()?); // TxType
+                        },
+                    }
+                }
+
+                let tx_type = tx_type.ok_or_else(|| de::Error::missing_field("transaction_type"))?;
+                let action = action.ok_or_else(|| de::Error::missing_field("action"))?;
+                let action: Action = match tx_type {
+                    TxType::Call    => Action::Call(de_value!(action)?),
+                    TxType::Create  => Action::Create(de_value!(action)?),
+                    TxType::Suicide => Action::Suicide(de_value!(action)?),
+                    TxType::Reward  => Action::Reward(de_value!(action)?)
+                };
+                let result = result.ok_or_else(|| de::Error::missing_field("result"))?;
+                let trace_address = trace_address.ok_or_else(|| de::Error::missing_field("trace_address"))?;
+                let subtraces = subtraces.ok_or_else(|| de::Error::missing_field("subtraces"))?;
+
+                Ok( TransactionTrace { trace_address, subtraces, action, result } )
+            }
+        }
+
+        const FIELDS: &'static [&'static str] = &["subtraces", "trace_address", "action", "type", "error", "result" ];
```

Comment:
Same as with other deserialize, can't we use serde-derived deserialization?
Maybe add a helper struct that matches the JSON format (with derived Deserialization) and then custom deserializer for this one would be just converting one representation to another?